### PR TITLE
debug(spool): Add metrics and report an error

### PR DESCRIPTION
**Pull request by olksdr**
Let's track the timings how long the envelope took to unspool. 
Also add error if the envelope was spooled longer than 2 hours.

Note: This can be chatty in case of the long lasting incident, but this way we will be able to see if any of the envelopes are stuck in the spool, and cannot be unspooled anymore. 
  
This is temporary and just to help with debugging of the on-disk spool. 

related to https://github.com/getsentry/team-ingest/issues/257

#skip-changelog

### CODE REVIEW

File: relay-server/src/actors/spooler/mod.rs [hunk 1]
```diff
@@ -415,9 +416,26 @@ impl OnDisk {
...
+
+        // If envelope stays longer than an 2 hours in on-disk spool, produce an error, so we could
+        // debug what kind of envelope is that.
+        if start_time_instant.elapsed() > Duration::from_secs(7200) {
+            relay_log::error!(
+                tags.project_key = %envelope.meta().public_key(),
+                tags.project_id = %envelope.meta().project_id().unwrap_or(ProjectId::new(0)),
+                event_id = ?envelope.event_id(),
+                duration = ?start_time_instant.elapsed(),
+                "Spooled envelope spent more than 1 hour in on-disk spool"
```
Comment (Dav1dde) [inline relay-server/src/actors/spooler/mod.rs]:
```suggestion
                "Spooled envelope spent more than 2 hours in on-disk spool"
```

**Review by Dav1dde — APPROVED** [review]

File: relay-server/src/statsd.rs [hunk 2]
```diff
@@ -243,6 +243,8 @@ pub enum RelayTimers {
     /// Total time in milliseconds an envelope spends in Relay from the time it is received until it
     /// finishes processing and has been submitted to the upstream.
     EnvelopeTotalTime,
+    /// Total time in milliseconds and envelope spends in the Relay on disk buffer.
```
Comment (iker-barriocanal) [inline relay-server/src/statsd.rs]:
```suggestion
    /// Total time in milliseconds an envelope spends in Relay on the disk buffer.
```

**Review by iker-barriocanal — APPROVED** [review]

File: relay-server/src/actors/spooler/mod.rs [hunk 3]
```diff
@@ -415,9 +416,26 @@ impl OnDisk {
...
 
-        envelope.set_start_time(start_time.into_inner());
+        // This metric should show how long we waited till the envelope was unspooled.
+        relay_statsd::metric!(
+            timer(RelayTimers::EnvelopeOnDiskBufferTime) = start_time_instant.elapsed()
+        );
+
+        // If envelope stays longer than an 2 hours in on-disk spool, produce an error, so we could
+        // debug what kind of envelope is that.
+        if start_time_instant.elapsed() > Duration::from_secs(7200) {
```
Comment (iker-barriocanal) [inline relay-server/src/actors/spooler/mod.rs]: nit: this should allow removing the comment?
```suggestion
        if start_time_instant.elapsed() > Duration::from_secs(3600 * 2) {
```

Comment (Dav1dde) [reply relay-server/src/actors/spooler/mod.rs]: I like the comment for the reasoning, less for that 7200 = 2 hours

File: relay-server/src/actors/spooler/mod.rs [hunk 4]
```diff
@@ -415,9 +416,26 @@ impl OnDisk {
...
+        let start_time_instant = StartTime::from_timestamp_millis(received_at as u64).into_inner();
 
-        envelope.set_start_time(start_time.into_inner());
+        // This metric should show how long we waited till the envelope was unspooled.
+        relay_statsd::metric!(
+            timer(RelayTimers::EnvelopeOnDiskBufferTime) = start_time_instant.elapsed()
+        );
+
+        // If envelope stays longer than an 2 hours in on-disk spool, produce an error, so we could
+        // debug what kind of envelope is that.
```
Comment (TBS1996) [inline relay-server/src/actors/spooler/mod.rs]:
```suggestion
        // If an envelope stays longer than 2 hours in on-disk spool, produce an error, so we could
        // debug what kind of envelope it is.
```

**Review by TBS1996 — APPROVED** [review]

**Pull request merged**